README.md

```diff
@@ -0,0 +1,185 @@
+1. It can struggle to bind more exotic types beyond primitives from the CLR and the most common collection types.
+
+This library provides an alternative approach to options binding for FSharp developers in the form of declarative computation expressions. Rather than relying on reflection magic it gives you control over the whole options binding process in a composable and type safe way. It provides first class support for `Option` values. It models the outcome of the bind operation with a `BindResult` which can be either `Success` or `Failure` and it reports as many issues as it can when a `Failure` occurs so you can fix them all in one go.
+
+## Build Status
+
+[![Build history](https://buildstats.info/github/chart/SymbolicaDev/Symbolica.Extensions.Configuration.FSharp?branch=master)](https://github.com/SymbolicaDev/Symbolica.Extensions.Configuration.FSharp/actions)
+
+## Installation
+
+[![NuGet Badge](https://buildstats.info/nuget/Symbolica.Extensions.Configuration.FSharp)](https://www.nuget.org/packages/Symbolica.Extensions.Configuration.FSharp/)
+
+Using the `dotnet` cli
+
+```sh
+dotnet add package Symbolica.Extensions.Configuration.FSharp
+```
+
+or with the NuGet package manager
+
+```sh
+PM> Install-Package Symbolica.Extensions.Configuration.FSharp
+```
+
+## Usage
+
+The primary means of using this library is through two computation expressions called `section` and `optSection`. These provide a declarative DSL for safely binding a type from an `IConfiguration` instance.
+
+## Example
+
+Imagine you have the following records defined to model some options
+
+```fsharp
+type SubOptions =
+    { OptionalNumber: float option
+      Bool: bool }
+
+type Options =
+    { Name: string
+      Count: int
+      SubOptions: SubOptions
+      OptSubOptions: SubOptions option }
+```
+
+We can bind this from an `IConfiguration` like this
+
+```fsharp
+let bindOptions (config: IConfiguration) =
+    section "Options" {
+        let! name = value "Name"
+        and! count = value "Count"
+
+        and! subOptions =
+            section "Sub" {
+                let! optionalNumber = optValueOf Decode.float "MaybeDecimal"
+                and! bool = valueOf Decode.bool "bool"
+
+                return
+                    { OptionalNumber = optionalNumber
+                      Bool = bool }
+            }
+
+        and! optSubOptions =
+            optSection "OptSub" {
+                let! optionalNumber = optValueOf Decode.float "MaybeDecimal"
+                and! bool = valueOf Decode.bool "bool"
+
+                return
+                    { OptionalNumber = optionalNumber
+                      Bool = bool }
+            }
+
+        return
+            { Name = name
+              Count = count
+              SubOptions = subOptions
+              OptSubOptions = optSubOptions }
+    }
+    |> Binder.eval config
+```
```

Comment:
Obviously I don't really speak this language well, but do all the `let!` etc appear because you lifted this example from somewhere `async` that isn't shown here? If so, maybe get rid of the `!` to minimize the example somewhat.

Reply:
No the `let!` is a monadic bind, so in this case it's syntactic sugar for `Binder.bind`. As in, I want to assign this value to whatever's inside the Binder. So it would be the value of the config, assuming it's present and correct. `let!` is more like a programmable async/await for a completely different use case here.